flatsurf/geometry/matrix_2x2.py

```diff
@@ -22,87 +22,14 @@
 #  along with sage-flatsurf. If not, see <https://www.gnu.org/licenses/>.
 ######################################################################
 from __future__ import absolute_import, print_function, division
-from six.moves import range, map, filter, zip
 
-from sage.misc.cachefunc import cached_function
+from sage.rings.all import AA, QQbar, RR
 
-from sage.rings.all import ZZ, QQ, AA, QQbar, RR, CC, RDF, CDF, RIF, CIF
-from sage.rings.rational import Rational
-from sage.rings.complex_interval_field import ComplexIntervalField
-
-from math import pi as pi_float
-
-from sage.symbolic.constants import pi
 from sage.matrix.constructor import matrix, identity_matrix
 from sage.modules.free_module_element import vector
 
-def number_field_to_AA(a):
-    r"""
-    It is a mess to convert an element of a number field to the algebraic field
-    ``AA``. This is a temporary fix.
-    """
-    try:
-        return AA(a)
-    except TypeError:
-        return AA.polynomial_root(a.minpoly(), RIF(a))
 
-def is_similarity(m):
-    r"""
-    Return ``True`` if ``m`` is a similarity and ``False`` otherwise.
-
-    EXAMPLES::
-
-        sage: from flatsurf.geometry.matrix_2x2 import is_similarity
-
-        sage: is_similarity(matrix([[0,1],[1,0]]))
-        True
-        sage: is_similarity(matrix([[0,-2],[2,0]]))
-        True
-        sage: is_similarity(matrix([[1,1],[0,1]]))
-        False
-    """
-    n = m * m.transpose()
-    return n[0,1].is_zero() and n[1,0].is_zero()
-
-def homothety_rotation_decomposition(m):
-    r"""
-    Return a couple composed of the homothety and a rotation matrix.
-
-    The coefficients of the returned pair are either in the ground field of
-    ``m`` or in the algebraic field ``AA``.
-
-    EXAMPLES::
-
-        sage: from flatsurf.geometry.matrix_2x2 import homothety_rotation_decomposition
-
-        sage: R.<x> = PolynomialRing(QQ)
-        sage: K.<sqrt2> = NumberField(x^2 - 2, embedding=1.4142)
-        sage: m = matrix([[sqrt2, -sqrt2],[sqrt2,sqrt2]])
-        sage: a,rot = homothety_rotation_decomposition(m)
-        sage: a
-        2
-        sage: rot
-        [ 1/2*sqrt2 -1/2*sqrt2]
-        [ 1/2*sqrt2  1/2*sqrt2]
-    """
-    if not is_similarity(m):
-        raise ValueError("the matrix must be a similarity")
-
-    det = m.det()
-
-    if not det.is_square():
-        if not AA.has_coerce_map_from(m.base_ring()):
-            l = map(number_field_to_AA,m.list())
-            M = MatrixSpace(AA,2)
```

Comment:
Note that `MatrixSpace` is not defined. So this code path would not have worked.